usingnamespace problem with imported const entries

```zig
usingnamespace struct {
    pub const foo = 42;
};

comptime {
    @compileLog(foo);
}
```

```
usens.zig:2:9: error: redefinition of 'foo'
    pub const foo = 42;
        ^
usens.zig:2:9: note: previous definition is here
    pub const foo = 42;
        ^
usens.zig:6:17: note: referenced here
    @compileLog(foo);
```

This blocks #3311 as I'm using this construct to conditionally define some constants.

--- a/src/analyze.cpp
+++ b/src/analyze.cpp
@@ -3676,7 +3676,7 @@ ZigVar *add_variable(CodeGen *g, AstNode *source_node, Scope *parent_scope, Buf 
                 }
                 if (search_scope != nullptr) {
                     Tld *tld = find_decl(g, search_scope, name);
-                    if (tld != nullptr) {
+                    if (tld != nullptr && tld != src_tld) {
                         ErrorMsg *msg = add_node_error(g, source_node,
                                 buf_sprintf("redefinition of '%s'", buf_ptr(name)));
                         add_error_note(g, msg, tld->source_node, buf_sprintf("previous definition is here"));

--- a/test/stage1/behavior/usingnamespace.zig
+++ b/test/stage1/behavior/usingnamespace.zig
@@ -12,3 +12,11 @@ test "usingnamespace inside a generic struct" {
     std2.testing.expect(true);
     testing2.expect(true);
 }
+
+usingnamespace struct {
+    pub const foo = 42;
+};
+
+test "usingnamespace does not redeclare an imported variable" {
+    comptime std.testing.expect(foo == 42);
+}

